src/briefcase/platforms/android/apk.py

```diff
@@ -1 +1,205 @@
+                    download_path=Path.home() / ".briefcase" / "tools",
+                )
+            except requests_exceptions.ConnectionError:
+                raise NetworkFailure("downloading Android SDK")
+            with ZipFile(android_sdk_zip_path) as android_sdk_zip:
+                android_sdk_zip.extractall(path=self.android_sdk_path_tmp)
+            # Remove the ZIP file; it has no purpose now that it is extracted.
+            android_sdk_zip_path.unlink()
+            # Set executable permissions; Python ZipFile ignores these, but we need them.
+            for binpath in (self.android_sdk_path_tmp / "tools" / "bin").glob("*"):
+                binpath.chmod(0o755)
+            self.android_sdk_path_tmp.rename(self.android_sdk_path)
+
+        print("Ensuring all Android SDK licenses are accepted...")
+        self.subprocess.run(
+            [self.android_sdk_path / "tools" / "bin" / "sdkmanager", "--licenses"],
+            check=True,
+            cwd=self.android_sdk_path,
+        )
+
+    def build_app(self, app: BaseConfig, **kwargs):
+        """
+        Build an application.
+
+        :param app: The application to build
+        """
+        print()
+        print("[{app.app_name}] Building Android APK...".format(app=app))
+
+        try:
+            print()
+            # Use `gradle` to build.
+            env = self.os.environ.copy()
+            env["ANDROID_SDK_ROOT"] = str(self.android_sdk_path)
+            appdir_path = self.bundle_path(app)
+            self.subprocess.run(
+                [appdir_path / "gradlew", "assembleDebug",],
+                env=env,
+                check=True,
+                cwd=str(appdir_path),
+            )
+
+            # Make the binary executable.
+            self.os.chmod(str(self.binary_path(app)), 0o755)
+        except subprocess.CalledProcessError:
+            print()
+            raise BriefcaseCommandError(
+                "Error while building app {app.app_name}.".format(app=app)
+            )
+
+
+class ApkRunCommand(ApkMixin, RunCommand):
+    description = "Run a Linux AppImage."
+
+    def run_app(self, app: BaseConfig, **kwargs):
+        """
+        Start the application.
+
+        :param app: The config object for the app
+        :param base_path: The path to the project directory.
+        """
+        raise NotImplementedError()
+        # TODO: Respect `-d`
+        #
+        # Find all possible targets:
+        # - running devices `adb list-devices`
+        # - available emulator devices (`ls ~/.android/emulator`)
+        # - mention `briefcase auto-generated device` even if we haven't even created it yet
+        #
+        # Show all possible targets
+        #
+        # If you picked an emulator device but it's not started, start it
+        #
+        # If you picked the briefcase auto-generated device, then create it :)
+
+
+class ApkPackageCommand(ApkMixin, PackageCommand):
+    description = "Package an Android APK."
```

Comment:
Is it okay if `package` and `build` are the same for Android?

I'll keep this a no-op like this, per https://github.com/beeware/briefcase/blob/master/src/briefcase/platforms/iOS/xcode.py , but figured I'd check.

Reply:
At least for first pass, it's fine for package to be a no-op. Longer term, package would be where APK signing would occur.

Reply:
Gotcha -- makes sense that `package` would involve signing.